src/lib/statistics/statistics_objects/range_filter.cpp

```diff
@@ -0,0 +1,288 @@
+#include "range_filter.hpp"
+
+#include <algorithm>
+#include <memory>
+#include <optional>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "abstract_statistics_object.hpp"
+#include "resolve_type.hpp"
+#include "statistics/statistics_objects/min_max_filter.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+template <typename T>
+RangeFilter<T>::RangeFilter(std::vector<std::pair<T, T>> ranges)
+    : AbstractStatisticsObject(data_type_from_type<T>()), ranges(std::move(ranges)) {}
+
+template <typename T>
+Cardinality RangeFilter<T>::estimate_cardinality(const PredicateCondition predicate_condition,
+                                                 const AllTypeVariant& variant_value,
+                                                 const std::optional<AllTypeVariant>& variant_value2) const {
+  return 0;
```

Comment:
When would someone reasonably ask the RangeFilter for a cardinality estimation? Simply returning a random number does not feel right.